src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java

```diff
@@ -0,0 +1,281 @@
+
+        executeBlocking(totalNumberOfUpdatedCartDiscounts);
+        assertThat(totalNumberOfUpdatedCartDiscounts).isCompletedWithValue(NUMBER_OF_RESOURCE_UNDER_TEST);
+
+        // Assert actual state of CTP project (total number of existing cart discounts)
+        final CompletableFuture<Integer> totalNumberOfCartDiscounts =
+            CTP_TARGET_CLIENT.execute(CartDiscountQuery.of())
+                             .thenApply(PagedQueryResult::getTotal)
+                             .thenApply(Long::intValue)
+                             .toCompletableFuture();
+        executeBlocking(totalNumberOfCartDiscounts);
+        assertThat(totalNumberOfCartDiscounts).isCompletedWithValue(NUMBER_OF_RESOURCE_UNDER_TEST);
+
+        // Assert statistics
+        assertThat(syncStatistics).hasValues(NUMBER_OF_RESOURCE_UNDER_TEST, 0, NUMBER_OF_RESOURCE_UNDER_TEST, 0);
+
+        assertThat(errorCallBackExceptions).isEmpty();
+        assertThat(errorCallBackMessages).isEmpty();
+        assertThat(warningCallBackMessages).isEmpty();
+
+        saveNewResult(CART_DISCOUNT_SYNC, UPDATES_ONLY, totalTime);
+    }
+
+    @Test
+    void sync_WithSomeExistingCartDiscounts_ShouldSyncCartDiscounts() throws IOException {
+        // preparation
+        final List<CartDiscountDraft> cartDiscountDrafts = buildCartDiscountDrafts(NUMBER_OF_RESOURCE_UNDER_TEST);
+        final int halfNumberOfDrafts = cartDiscountDrafts.size() / 2;
+        final List<CartDiscountDraft> firstHalf = cartDiscountDrafts.subList(0, halfNumberOfDrafts);
+
+        // Create first half of cart discount drafts to target project
+        // with different stacking mode (STOP_AFTER_THIS_DISCOUNT)
+        CompletableFuture.allOf(
+                firstHalf.stream()
+                        .map(CartDiscountDraftBuilder::of)
+                        .map(builder -> builder.stackingMode(StackingMode.STOP_AFTER_THIS_DISCOUNT))
+                        .map(CartDiscountDraftBuilder::build)
+                        .map(draft -> CTP_TARGET_CLIENT.execute(CartDiscountCreateCommand.of(draft)))
+                        .map(CompletionStage::toCompletableFuture)
+                        .toArray(CompletableFuture[]::new))
+                .join();
+
+        final CartDiscountSync cartDiscountSync = new CartDiscountSync(cartDiscountSyncOptions);
+
+
+        // benchmark
+        final long beforeSyncTime = System.currentTimeMillis();
+        // Create half of the cart discount drafts to target project
+        // and update half of the cart discount drafts to target project with different stacking mode (STACKING)
+        final CartDiscountSyncStatistics syncStatistics = executeBlocking(cartDiscountSync.sync(cartDiscountDrafts));
+        final long totalTime = System.currentTimeMillis() - beforeSyncTime;
+
+        // assert on threshold (based on history of benchmarks; highest was ~12 seconds)
+        final int threshold = 24000; // double of the highest benchmark
```

Comment:
I guess this is not true for this benchmark yet ;)

Reply:
50527df